File: solutions/go/hamming/1/hamming.go
package hamming

import "errors"

func Distance(a, b string) (int, error) {
    var distance int
    if len(a) != len(b) {
        return distance, errors.New("inputs must have same length")
    }    
	for i, _ := range a {
        if a[i] != b[i] {
            distance++
        }
    }    
    return distance, nil
}

File: solutions/csharp/hamming/1/Hamming.cs
public static class Hamming
{
    public static int Distance(string fs, string ss)
    {
        if (fs.Length != ss.Length)
            throw new ArgumentException();
        var distance = 0;
        for (var i = 0; i < fs.Length; i++)
        {
            if (fs[i] != ss[i])
                distance++;
        }
        return distance;
    }
}